src/nvim/tui/tui.c

```diff
@@ -1268,7 +1268,7 @@ static void patch_terminfo_bugs(TUIData *data, const char *term,
   bool teraterm = terminfo_is_term_family(term, "teraterm");
   bool putty = terminfo_is_term_family(term, "putty");
   bool screen = terminfo_is_term_family(term, "screen");
-  bool tmux = terminfo_is_term_family(term, "tmux");
+  bool tmux = terminfo_is_term_family(term, "tmux") || !!os_getenv("TMUX");
```

Comment:
well, it is no that reliable as it can be thought.
It is some cases where users pass `TMUX` variable to shells to avoid some bugs.

By the way, there is also `TMUX_PANE` variable, which have lesser chance to be used for such hacks, but it is anyway very bad way to detect because of the said reason 😼